pkg/packageinstall/packageinstall.go

```diff
@@ -374,6 +374,10 @@ func (pi *PackageInstallCR) reconcileDelete(modelStatus *reconciler.Status) (rec
 		existingApp.Spec.Canceled = pi.model.Spec.Canceled
 	}
 
+	if existingApp.Spec.InstallationNamespace != pi.model.Spec.InstallationNamespace {
```

Comment:
I'm not saying to change this here in this PR. Just observing and wondering why we even have these checks? Sort of feels redundant, they could just be assignments because if the value is different, we change it to `pi.model.Spec` and it's not then it must be equal to `pi.model.Spec`. So why even bother doing checks....

Reply:
Yeah, the check seems weird. For now, I have removed the check for the defaultNamespace. If we want we can update the others in a separate PR.